### PR TITLE
Implement Unified Embed for Tweet URLs

### PR DESCRIPTION
**Pull request by msarit**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for Tweet embeds (not to be confused with TwitterTimeline embeds).

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15581

## QA Instructions, Screenshots, Recordings
Example Tweet URLs:
https://twitter.com/aritdeveloper/status/1483614684884484099
https://twitter.com/aritdeveloper/status/1483775972688240643

Example Tweet IDs
1483614684884484099
1483775972688240643

- As a user, create a Tweet liquid tag using this format: `{% embed <tweet-url> %}`. The Liquid Tag should render properly.
- Also create Tweet liquid tag using the old methods: `{% tweet <tweet-id> %}` and `{% twitter <tweet-id> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?
None

## Added/updated tests?
- [X] Yes

## [Forem core team only] How will this change be communicated?
- [X] I will share this change internally with the appropriate teams
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.

## [optional] Are there any post deployment tasks we need to perform?
None

### CODE REVIEW

**Review by msarit — COMMENTED** [review]
I need help.

For some reason, the author picture in the Tweet embed is broken. When inspecting the embed in my local, the `src` for the image is:
```
https://res.cloudinary.com/Optional/image/fetch/s--rfDjWf2Z--/c_limit,f_auto,fl_progressive,Cq_auto,w_880/https://pbs.twimg.com/profile_images/1445880121945169921/zBfcgvo8_normal.jpg
```
Looks very strange to me. In the view for the Tweet liquid tag, the author-image `src` is rendered thus:

```ruby
<%= tweet.full_fetched_object_serialized[:user][:profile_image_url_https] %>
```

I would appreciate some debugging help! 🙏🏾

**Review by jgaskins — APPROVED** [review]
Works great! Looks great! #crushingit

**Comment by jgaskins** [comment]
> For some reason, the author picture in the Tweet embed is broken.

Hmm, I noticed this on mine, too, but assumed that was a config issue locally for me.

**Comment by jgaskins** [comment]
@msarit Oh, this _is_ just a config issue. Cloudinary configuration is probably in your `.env` file. If you comment those out it should fall back to using Carrierwave local-storage settings.

File: app/liquid_tags/tweet_tag.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class TweetTag < LiquidTagBase
   include ActionView::Helpers::AssetTagHelper
   PARTIAL = "liquids/tweet".freeze
-  ID_REGEXP = /\A\d{10,20}\z/ # id must be all numbers between 10 and 20 chars
+  REGISTRY_REGEXP = %r{https://twitter.com/\w{1,15}/status/(?<id>\d{10,20})}
+  VALID_ID_REGEXP = /\A(?<id>\d{10,20})\Z/
+  REGEXP_OPTIONS = [REGISTRY_REGEXP, VALID_ID_REGEXP].freeze
```
Comment (djuber) [inline app/liquid_tags/tweet_tag.rb:6]: I'm not super familiar with the internals of this class - is REGEXP_OPTIONS used in a way that _any_ 10-20 digit string is construed as a valid twitter id - and do we have any other (youtube?) embeds with overlapping recognizers?

if I have an 11 digit "tweet" id, and an 11 digit youtube video id, 
  VALID_ID_REGEXP = /\A(?<video_id>[a-zA-Z0-9_-]{11})

which one, if any, does `{% embed 12345678901 %}` choose?

I'll _test_ that in your branch but it's the first thing I saw reading this code.

Comment (msarit) [reply app/liquid_tags/tweet_tag.rb:6]: good question, I will look into it

Comment (djuber) [reply app/liquid_tags/tweet_tag.rb:6]: answer: no concern here, tweet and youtube tags may take ids but embed does not

Comment (djuber) [reply app/liquid_tags/tweet_tag.rb:6]: it looks like I need an api key to fetch a tweet in the api? 

I don't see that in .env so I assume I need to set that up in config?

Comment (msarit) [reply app/liquid_tags/tweet_tag.rb:6]: ahh yes @djuber I needed to grab keys from developer.twitter.com and plug them into `config`

Comment (msarit) [reply app/liquid_tags/tweet_tag.rb:6]: So, @djuber, you brought up a very good point about the `embed` keyword returning an unexpected embed if the `id` say matches another tag.

For liquid tags that receive just a number(`id`), the keyword is the type of embed expected (e.g. `{% twitter 12345 %}`. So I don't think a mismatched embed will result.

For liquid tags that use the `embed` keyword, the expected input is a URL, which specifies the type of embed to be returned. By the time the `id` is parsed out, the correct embed type has already been chosen from the Registry. So again I don't think a mismatched embed will result.

Let me know if my understanding has any holes. 😅

**Review by jeremyf — APPROVED** [review]

**Pull request merged**